snooty/parser.py

```diff
@@ -700,138 +705,150 @@ def create_page() -> Tuple[Page, EmbeddedRstParser]:
                 )
                 return doc
 
-        elif name == "literalinclude":
-            if argument_text is None:
-                self.diagnostics.append(ExpectedPathArg(name, line))
-                return doc
-
-            _, filepath = util.reroot_path(
-                FileId(argument_text), self.docpath, self.project_config.source_path
-            )
-
-            # Attempt to read the literally included file
-            try:
-                text = filepath.read_text(encoding="utf-8")
-            except OSError as err:
-                self.diagnostics.append(
-                    CannotOpenFile(argument_text, err.strerror, line)
+        elif name == "literalinclude" or name == "input" or name == "output":
+            if name == "literalinclude":
+                if argument_text is None:
+                    self.diagnostics.append(ExpectedPathArg(name, line))
+                    return doc
+            if argument_text is not None:
```

Comment:
If I'm scanning this correctly, there's no else clause here, correct? Could we do e.g. `if argument_text is None: return doc` and dedent the `if` body?

Reply:
Yep!